mapie/tests/test_quantile_regression.py

```diff
@@ -519,6 +519,151 @@ def test_linear_regression_results(strategy: str) -> None:
+    cv="prefit".
+    """
+    with pytest.raises(
+        ValueError,
+        match=r".*You need to have provided 3 different estimators, th*",
+    ):
+        gb_trained1, gb_trained2 = clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        list_estimators = [gb_trained1, gb_trained2]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit"
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_prefit_no_fit_predict() -> None:
+    """
+    Check that the estimators given have a prefit and fit attribute.
+    """
+    with pytest.raises(
+        ValueError,
+        match=r"Invalid estimator. Please provide a regressor with fit and*",
+    ):
+        gb_trained1, gb_trained2 = clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        gb_trained3 = 3
+        list_estimators = [gb_trained1, gb_trained2, gb_trained3]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit",
+            alpha=0.3
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_non_trained_estimator() -> None:
+    """
+    Check that the estimators are all already trained when used in prefit.
+    """
+    with pytest.raises(
+        ValueError,
+        match=r".*instance is not fitted yet. Call 'fit' with appropriate*",
+    ):
+        gb_trained1, gb_trained2, gb_trained3 = clone(gb), clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        list_estimators = [gb_trained1, gb_trained2, gb_trained3]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit",
+            alpha=0.3
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_warning_alpha_prefit() -> None:
+    """
+    Check that the user is warned that the alphas need to be correctly set.
+    """
+    with pytest.warns(
+        UserWarning,
+        match=r".*WARNING: The alpha that is set needs to be the same*"
+    ):
+        gb_trained1, gb_trained2, gb_trained3 = clone(gb), clone(gb), clone(gb)
+        gb_trained1.fit(X_train, y_train)
+        gb_trained2.fit(X_train, y_train)
+        gb_trained3.fit(X_train, y_train)
+        list_estimators = [gb_trained1, gb_trained2, gb_trained3]
+        mapie_reg = MapieQuantileRegressor(
+            estimator=list_estimators,
+            cv="prefit",
+            alpha=0.3
+        )
+        mapie_reg.fit(
+            X_calib,
+            y_calib
+        )
+
+
+def test_prefit_and_non_prefit_equal() -> None:
+    """
+    Check that the user is warned that the alphas need to be correctly set.
+    """
+    list_estimators = []
+    alphas_ = [0.15, 0.85, 0.5]
```

Comment:
Possible to parametrize `alpha` (and test 2-3 values) and deduce the `alphas_` accordingly ?

Reply:
done